src/tool/subcommands/api_cmd.rs

```diff
@@ -164,42 +167,105 @@ impl ApiCommands {
             Self::Compare {
                 forest: UrlFromMultiAddr(forest),
                 lotus: UrlFromMultiAddr(lotus),
-                snapshot_files,
                 filter,
                 filter_file,
                 fail_fast,
-                n_tipsets,
                 run_ignored,
                 max_concurrent_requests,
-                miner_address,
-                worker_address,
-                eth_chain_id,
+                create_tests_args,
             } => {
-                let config = ApiTestFlags {
-                    filter,
+                let forest = rpc::Client::from_url(forest);
+                let lotus = rpc::Client::from_url(lotus);
+
+                let tests = create_tests(create_tests_args)?;
+                run_tests(
+                    tests,
+                    forest,
+                    lotus,
+                    max_concurrent_requests,
                     filter_file,
-                    fail_fast,
-                    n_tipsets,
+                    filter,
                     run_ignored,
-                    max_concurrent_requests,
-                    miner_address,
-                    worker_address,
-                    eth_chain_id,
-                };
-
-                compare_apis(
-                    rpc::Client::from_url(forest),
-                    rpc::Client::from_url(lotus),
-                    snapshot_files,
-                    config,
+                    fail_fast,
                 )
                 .await?
             }
+            Self::DumpTests {
+                create_tests_args,
+                path,
+            } => {
+                for RpcTest {
```

Comment:
I would dump `create_tests_args` and `path` as well since it's critical to have forest and lotus load the proper snapshot before running the generated tests

Reply:
Moved to accept a `--path` parameter